opentelemetry/src/baggage.rs

```diff
@@ -412,7 +412,7 @@ impl BaggageExt for Context {
     }
 
     fn with_cleared_baggage(&self) -> Self {
-        self.with_value(Baggage::new())
+        self.with_value(BaggageContextValue(Baggage::new()))
```

Comment:
nit: this looks slightly simpler

```suggestion
        self.with_baggage(Baggage::new())
```